scripts/knot_benchmark/src/benchmark/cases.py

```diff
@@ -0,0 +1,223 @@
+from __future__ import annotations
+
+import abc
+import enum
+import os
+import shutil
+from pathlib import Path
+import subprocess
+import sys
+import logging
+
+from benchmark import Command
+from benchmark.projects import Project
+
+
+class Benchmark(enum.Enum):
+    """Enumeration of the benchmarks to run."""
+
+    COLD = "cold"
+    """Cold check of an entire project without a cache present."""
+
+    WARM = "warm"
+    """Re-checking the entire project without any changes"."""
+
+    INCREMENTAL = "incremental"
+    """Incremental check between two commits."""
```

Comment:
Not sure if we should remove this one again. I don't have an incremental benchmark yet but the idea would be to switch between two revisions.